build-logic-settings/build-environment/src/main/kotlin/gradlebuild/basics/BuildEnvironmentExtension.kt

```diff
@@ -24,4 +24,12 @@ interface BuildEnvironmentExtension {
     val gitCommitId: Property<String>
     val gitBranch: Property<String>
     val repoRoot: DirectoryProperty
+    val rootProjectBuildDir: DirectoryProperty
+    val artifactoryUserName: Property<String>
+    val artifactoryPassword: Property<String>
```

Comment:
❓ Can we avoid storing passwords in Properties here?

Reply:
I guess we would need to store them in env variables then here:
https://github.com/gradle/gradle/blob/ae0e9c8bc0a91d3a95872935d065069fd9cc6607/build-logic/publishing/src/main/kotlin/gradlebuild.publish-public-libraries.gradle.kts#L32-L36

Is that what you meant?

Reply:
The password is already available in an environment variable https://github.com/gradle/gradle/blob/c98cd71d84c06c9e12a0684863f09bb257ae7d2f/.teamcity/src/main/kotlin/promotion/PromotionProject.kt#L40

AFAICT we could reference that as we only need the password if we actually publish, correct me if I'm wrong @blindpirate